libqfieldsync/project_checker.py

```diff
@@ -439,3 +444,70 @@ def check_external_layers(
             )
 
         return None
+
+    def check_layer_geopackage_actions(
+        self, layer: LayerSource
+    ) -> Optional[FeedbackResult]:
+        """Check if layers from the same GeoPackage have offline and copy actions."""
+        if not layer.is_file or not layer.filename:
+            return None
+
+        # Check if is GeoPackage layer
+        if not layer.filename.lower().endswith(".gpkg"):
+            return None
+
+        current_action = layer.action
+        if current_action not in [SyncAction.OFFLINE, SyncAction.COPY]:
+            return None
+
+        # Group all layers by their GeoPackage file
+        gpkg_layers = {}
+        for project_layer in self.project.mapLayers().values():
+            source = LayerSource(project_layer)
+            if not source.is_file or not source.filename:
+                continue
+
+            if not source.filename.lower().endswith(".gpkg"):
+                continue
+
+            if source.action not in [SyncAction.OFFLINE, SyncAction.COPY]:
+                continue
+
+            if source.filename not in gpkg_layers:
+                gpkg_layers[source.filename] = []
+            gpkg_layers[source.filename].append((project_layer.name(), source.action))
+
+        # Check for GeoPackages with mixed actions
+        if layer.filename in gpkg_layers and len(gpkg_layers[layer.filename]) > 1:
+            has_offline = any(
+                action == SyncAction.OFFLINE
+                for _, action in gpkg_layers[layer.filename]
+            )
+            has_copy = any(
+                action == SyncAction.COPY for _, action in gpkg_layers[layer.filename]
+            )
+
+            if has_offline and has_copy:
+                offline_layers = [
+                    name
+                    for name, action in gpkg_layers[layer.filename]
+                    if action == SyncAction.OFFLINE
+                ]
+                copy_layers = [
+                    name
+                    for name, action in gpkg_layers[layer.filename]
+                    if action == SyncAction.COPY
+                ]
+
+                message = self.tr(
+                    "Multiple layers from the same GeoPackage have conflicting sync actions: "
+                    "Offline editing layers: {}\n. "
+                    "Copy layers: {}\n. "
+                    "The information on offline editing layers will be overwritten."
+                ).format(
+                    ", ".join(f'"{layer}"' for layer in offline_layers),
+                    ", ".join(f'"{layer}"' for layer in copy_layers),
+                )
```

Comment:
```suggestion
                message = self.tr(
                    "Layers having th same file datasource have conflicting sync actions that might cause dataloss."
                    "Layers <layer1>, <layer2>, ... share the same file <filename1>." ....
                    "Layers <layer3>, <layer4>, ... share the same file <filename2>."
                )
```

This should greatly simplify the code for the message generation. Let the users know about which layers are problematic and they can see the actual actions later when they decide to reconfigure.